icepyx/core/APIformatting.py

```diff
@@ -142,25 +142,25 @@ def combine_params(*param_dicts):
 
     Returns
     -------
-    single dictionary of all input dictionaries combined
+    url string of all input dictionaries combined
```

Comment:
Discussion question: are there pros/cons to turning this into a url here versus leaving it as a dictionary? Is this a CMR/API change that it now requires as input as a url and can't accept the dictionary input?

Reply:
keeping it as a dictionary results in the parameters being [percent-encoded](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode) which breaks the CMR pattern capability (pattern wildcards get converted).  As a middle ground (if this might break some functionality), could encode some parameters but not anything for the CMR pattern matching.

Reply:
That makes sense. I haven't gone all the way through to see if the change will break anything (I want to play with some subsetting), but if it doesn't break, then we can use this formatting for all parameters. Should we use urllib for this, or keep it hard-coded for our use case?

Reply:
@tsutterley Have you actually been able to place an order (through icepyx) with these updates? I started going through the example notebook and got errors when I called `region_a.order_granules()` because of internal calls to `req_params` (it's trying to get a value from a dict, but `req_params` is now a string). I didn't dig deeper to see how many instances would need to be modified to handle the change.

Currently the order functions are only weakly tested once a week on master by a cron or in a PR to master (because of the need for a login), but you can trigger them in another build by adding "[nsidc_tests]" to the commit message. There aren't currently enough unit tests to tell you much without viewing the travis output, but it should at least tell you here that it was unable to order and download data for some reason.

Reply:
@JessicaS11 sorry about that!  yes combining everything to a string broke the verbose output of pages and broke when there would be multiple pages of granules.  I made the dictionary-to-string into a separate function.  I ran it all the way through and it should be working.  I also tried adding url encoding using either `urllib.parse` or `requests` but it ended up breaking the cookies needed to "stay logged in".  So for now the strings are still not encoded but I haven't found an issue.

Reply:
I like the updates to have the dict-to-string as a separate function. That's super good to know re `urllib.parse` and `requests` - a problem for another day then. Where is the granule_name parameter being submitted? When I ran `region_a.order_granules(verbose=True)`, which prints out the URL of the order request, I didn't see it as part of the order submission, and it also doesn't show up in CMR_params either. From looking at `query.py`, it seems like they're being essentially initialized and then set again but never actually added to the `CMRparams` dictionary. Let me know if you want line numbers for what I think is happening.